vespajlib/src/main/java/com/yahoo/tensor/Tensor.java

```diff
@@ -62,7 +62,11 @@ public interface Tensor {
     /** Returns whether this have any cells */
     default boolean isEmpty() { return size() == 0; }
 
-    /** Returns the number of cells in this */
+    /**
+     *  Returns the number of cells in this.
+     *  TODO Figure how to best return an int instead of a long
```

Comment:
I don't think an int is enough. We could have a sizeAsInt() which would fail if the size doesn't fit, but this could also be done on the receiving side?

Reply:
There is already a cast to int in quite a few places (not checked) already, so making a sizeAsInt might be worth it.

I think that 2G values in a single tensor will not be reached in a very long time. And when that is reached you will have to overcome all the limits in java and all its libraries. All collections operate with int sizes and arrays only accepts int as size and index.

Reply:
Yes, but this is an interface that could be backed by anything ...